elasticsearch/templates/statefulset.yaml

```diff
@@ -191,8 +191,16 @@ spec:
 {{ toYaml .Values.securityContext | indent 10 }}
         image: "{{ .Values.image }}:{{ .Values.imageTag }}"
         imagePullPolicy: "{{ .Values.imagePullPolicy }}"
+      {{- if .Values.readinessProbe.customProbe }}
         readinessProbe:
-{{ toYaml .Values.readinessProbe | indent 10 }}
```

Comment:
Moving this line to below the currently hardcoded `exec` part will allow all of the keys to be overridden without changing anything else. This change would make the entire readinessProbe configurable without needing to add new logic or values.

With the way that Helm/YAML works it will just take the last key that is specified. So it will allow a custom exec command or the ability to disable the exec command itself by setting it to a null value. I wouldn't recommend it but it still makes sense for this to be possible to override.

Reply:
Hi,
yeah I get the point, but the reason for this commit is to specify a httpGet or tcpProbe instead of a exec probe.

Reply:
Not recommend, but here is how you would replace the exec with a tcpSocket probe instead. To remove a key from the configuration you can set it to a `null` value (https://v2.helm.sh/docs/chart_template_guide/#deleting-a-default-key)

```
readinessProbe:
  exec: null
  tcpSocket:
    port: 9200
  initialDelaySeconds: 5
  periodSeconds: 10
```

I have tested and confirmed that this is working as expected. The only change made is moving this line to below the currently defined exec probe.

Reply:
Ah now I understand what you mean. But in my opinion this feels a bit dirty. I like my approach a bit more - it's just cleaner. And since you have to modify the chart in both cases, I would do it my way. 
But as fatmcgav mentioned, this needs to be discussed anyway by elastic.

Reply:
> But in my opinion this feels a bit dirty. I like my approach a bit more - it's just cleaner. And since you have to modify the chart in both cases, I would do it my way.

Using the documented way to remove a default key for Helm doesn't feel dirty to me. Adding extra templating logic (which will require adding new testing) and duplicating one of the values for this chart feels like the dirtier option. 

> But as fatmcgav mentioned, this needs to be discussed anyway by elastic.

"Discussed by Elastic" is this case roughly translated to "ping @Crazybus to take a look" :)

Reply:
okay, i reverted and moved the readinessProbe value to the bottom.